src/common/DecayCounter.h

```diff
@@ -132,23 +109,27 @@ class DecayCounter {
    * decay etc.
    */
 
-  void reset(utime_t now) {
-    last_decay = now;
+  void reset() {
+    last_decay = clock::now();
     val = delta = 0;
   }
 
-  void decay(utime_t now, const DecayRate &rate);
+protected:
+  void decay(const DecayRate &rate) const;
+
+private:
+  mutable double val = 0.0;           // value
+  mutable double delta = 0.0;         // delta since last decay
+  mutable double vel = 0.0;           // recent velocity
+  mutable time last_decay = clock::zero();   // time of last decay
```

Comment:
@batrick Just curious about one thing. Why declare get() and decay(...) with const and also val, delta, vel and last_decay with mutable? Why not just get rid of these const and mutable stuff?

Reply:
Some places in the code call into `DecayCounter::get` with a const reference.